spec/models/book_spec.rb

```diff
@@ -0,0 +1,87 @@
+require 'rails_helper'
+
+describe Book do
+
+    describe '#create' do
+        context 'When fields are complete' do
+            subject(:book) { create(:book) }
+
+            it do
+                should validate_presence_of(:genre)
+            end
+            
+            it do
+                should validate_presence_of(:author)
+            end
+            
+            it do
+                should validate_presence_of(:image)
+            end
+            
+            it do
+                should validate_presence_of(:title)
+            end
+            
+            it do
+                should validate_presence_of(:publisher)
+            end
+            
+            it do
+                should validate_presence_of(:year)
+            end
+            
+            it do
+                is_expected.to be_valid
+            end
+        end
+    
+
+    #     context 'When the genre is nil' do
```

Comment:
Also be careful with commented code, we tend to not leave any of them in the repository.